### PR TITLE
8279047: Remove expired flags in JDK 20

### PR DESCRIPTION
**Pull request by dholmes-ora**
Expired Flags in 20:

- FilterSpuriousWakeups
- MinInliningThreshold
- PrefetchFieldsAhead

No remaining usages in code or tests.

- UseHeavyMonitors  (expired in PRODUCT ONLY)

As this flag now only exists for debug builds it has to be a "develop" flag rather than product. There are then changes to two tests that use this flag, so that they only run on a debug VM.

- test/jdk/java/lang/Thread/virtual/HoldsLock.java

The second @run that uses UseHeavyMonitors is moved to a second test segment that only applies to the debug VM.

- test/jdk/java/util/concurrent/ConcurrentHashMap/MapLoops.java

Change the test section that uses UseHeavyMonitors to only run on a debug VM and remove the IgnoreUnrecognizedOptions flag. Note that the existing test logic would run the same test twice on product builds.

No documentation in the manpage exists for any of the newly expired flags.

Obsoleted flags in 20:

- ExtendedDTraceProbes

Documented in manpage so moved from Deprecated section to Obsolete section. There is special handling for messages about use of this flag so that won't be updated until the flag is actually obsoleted (JDK-8279913).

- UseContainerCpuShares
- PreferContainerQuotaForCPUCount
- AliasLevel

Undocumented.

Java manpage updates:
- Updates Java version to 20
- Moved ExtendedDTraceProbe info as previously mentioned
- Applied changes from the .1 file that had not been applied to the markdown source so that they were not lost (and note the .1 file was also missing changes from the markdown file that had not been propagated).
- Removed an unrepresentable character (the section symbol) that was not being generated into either the html or nroff file correctly

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.java.net/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8279047](https://bugs.openjdk.org/browse/JDK-8279047): Remove expired flags in JDK 20

### Reviewers
 * [Vladimir Kozlov](https://openjdk.java.net/census#kvn) (@vnkozlov - **Reviewer**)
 * [Kim Barrett](https://openjdk.java.net/census#kbarrett) (@kimbarrett - **Reviewer**)
 * [Alan Bateman](https://openjdk.java.net/census#alanb) (@AlanBateman - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9127/head:pull/9127` \
`$ git checkout pull/9127`

Update a local copy of the PR: \
`$ git checkout pull/9127` \
`$ git pull https://git.openjdk.org/jdk pull/9127/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9127`

View PR using the GUI difftool: \
`$ git pr show -t 9127`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9127.diff">https://git.openjdk.org/jdk/pull/9127.diff</a>

</details>

### CODE REVIEW

**Review by vnkozlov — APPROVED** [review]
Looks good.

**Review by kimbarrett — APPROVED** [review]
Looks good.

**Comment by dholmes-ora** [comment]
Thanks for the reviews @vnkozlov  and @kimbarrett !

File: test/jdk/java/lang/Thread/virtual/HoldsLock.java [hunk 1]
```diff
@@ -28,6 +28,14 @@
...
  * @run testng/othervm --enable-preview HoldsLock
+ */
+
+/**
+ * @test
+ * @summary Test Thread.holdsLock when lock held by carrier thread
+ * @requires vm.continuations & vm.debug
+ * @modules java.base/java.lang:+open
+ * @compile --enable-preview -source ${jdk.version} HoldsLock.java
  * @run testng/othervm --enable-preview -XX:+UseHeavyMonitors HoldsLock
```
Comment (AlanBateman) [inline test/jdk/java/lang/Thread/virtual/HoldsLock.java:39]: The test updates look okay, probably don't need to duplicate the summary tag when it's unchanged but what you have is okay.

**Review by AlanBateman — APPROVED** [review]

**Comment by dholmes-ora** [comment]
Thanks for the review @AlanBateman !

/integrate

**Pull request closed without merging**